lib/cloudscopes/globals.rb

```diff
@@ -3,12 +3,16 @@
 #
 
 def publish(samples)
-  raise "Not running in EC2, so won't publish!" unless File.executable?("/usr/bin/ec2metadata")
+  unless system("test -f /sys/hypervisor/uuid && test `head -c 3 /sys/hypervisor/uuid` = ec2")
```

Comment:
If you're going on testing `/sys` data, then Its probably better to do it in Ruby code and not for to the system shell. This of course relies on `/sys` being there, which is not true for very old systems - which granted we may not want to support anyway.

Regardless, I think its a good place for refactoring - if you're rewriting that part, please move the test to a function and use Ruby code to test `/sys/hypervisor/uuid`. Also, while we are on the subject, you may want to consider using `/sys/hypervisor/compilation/compiled_by` or `/sys/hypervisor/version/extra` instead of assuming things about how UUIDs are generated on EC2 - both of these options I feel provide a clearer identification of EC2. If you decide to stick to the UUID, I suggest documenting that this is the prescribed way to do this by Amazon and link to http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/identify_ec2_instances.html , and also - make the test case insensitive.